sklearn/tree/_criterion.pyx

```diff
@@ -199,6 +180,63 @@ cdef class Criterion:
                                  - (self.weighted_n_left /
                                     self.weighted_n_node_samples * impurity_left)))
 
+    cdef void set_sample_pointers(
+        self,
+        SIZE_t start,
+        SIZE_t end
+    ) nogil:
+        """Placeholder for a method which will set sample pointers in the criterion.
+
+        Parameters
+        ----------
+        start : SIZE_t
+            The first sample to be used on this node
```

Comment:
```suggestion
            The index of the first sample to be used on computation of criteria of the current node.
```

Same fix for 'end'. 

Also you can remove the docstring in the other downstream classes cuz this one describes the general gist.

Reply:
just pushed a commit for this!